### PR TITLE
Particles in edit

### PR DESCRIPTION
**Pull request by ericrius1**
The particle explorer now is part of the entities window, and appears as a tab when a user selects a particle entity from the entity list tab. Also, a user can now create a particle emitter from the edit spawner toolbar, just as they can for other entity types.

### CODE REVIEW

**Comment by ericrius1** [comment]
Asana: https://app.asana.com/0/31759584831097/80715240133020
Test: http://rawgit.com/ericrius1/hifi/particlesInEdit2/examples/edit.js

**Comment by hifi-gustavo** [comment]
Build console output for commit: f98941e80f0998de8464d0ba946cdb4c1c568d44 available at: https://gist.github.com/93ec3129c1f1fb90f8c7<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7110.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7110.dmg

**Comment by imgntn** [comment]
@ericrius1 the particle explorer tab doesn't disappear when i delete the particle system -- something with what's selected in the entity window?  particle explorer goes away when i select a new entity from the list

File: examples/edit.js [hunk 1]
```diff
@@ -1843,3 +1880,34 @@ propertyMenu.onSelectMenuItem = function(name) {
...
+entityListTool.webView.eventBridge.webEventReceived.connect(function(data) {
+    var data = JSON.parse(data);
+    if (data.type == "selectionUpdate") {
+        var ids = data.entityIds;
+        if(ids.length === 1) {
+            if (Entities.getEntityProperties(ids[0], "type").type === "ParticleEffect" ) {
+                if (JSON.stringify(selectedParticleEntity) !== JSON.stringify(ids[0])) {
+                    // We already had a particle entity selected, so destroy that one and create the new one
+                    particleExplorerTool.destroyWebView();
+                }
```
Comment (imgntn) [inline examples/edit.js]: i can feel this 2 second lag lag  --  there's already an event that gets sent over the bridge called 'page_loaded' when the web side is done --- there was already some code in there for this that you removed
 
-    this.onWebEventReceived = function(data) {
 -        var _data = JSON.parse(data);
 -        if (_data.messageType === 'page_loaded') {
 -            _this.webWindow.setVisible(true);
 -            _this.pageLoaded = true;
 -            sendInitialSettings(particleProperties);
 -        }
 -        if (_data.messageType === 'settings_update') {
 -            editEntity(_data.updatedSettings);
 -            return;
 -        }

**Comment by ericrius1** [comment]
@imgntn - thanks for pointing out those failure cases. They should be fixed now

**Comment by hifi-gustavo** [comment]
Build console output for commit: 46679b29f7c17ac86228816b95dd01df044a87f4 available at: https://gist.github.com/93ec3129c1f1fb90f8c7<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7110.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7110.dmg

File: examples/edit.js (hunk 1 above)
Comment (ericrius1) [inline examples/edit.js]: Good point- it's updated and now less lag!

**Comment by hifi-gustavo** [comment]
Build console output for commit: 46907f446b6c3a61b408adb3a463cd71f8f02ece available at: https://gist.github.com/93ec3129c1f1fb90f8c7<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7110.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7110.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: ad43e4ff6d4eb47c1bc14b49834fd680edc812c1 available at: https://gist.github.com/93ec3129c1f1fb90f8c7<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7110.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7110.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: bf21bc2d483e234bdfcf41a43b8214b11707591e available at: https://gist.github.com/93ec3129c1f1fb90f8c7<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7110.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7110.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: d9d0e78dc5b3e7b0375108a358061149f4b2acda available at: https://gist.github.com/93ec3129c1f1fb90f8c7<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7110.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7110.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 023fd0607df33e56d49e90b6098dc5e2edbf4c71 available at: https://gist.github.com/93ec3129c1f1fb90f8c7<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7110.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7110.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: c43500f38a437491b46ec7d3c698a0619518e622 available at: https://gist.github.com/93ec3129c1f1fb90f8c7<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7110.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7110.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: ce2d76d3f3dd4b274569d92ad3bf6e9ab75e353f available at: https://gist.github.com/93ec3129c1f1fb90f8c7<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7110.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7110.dmg

**Pull request merged**